src/messaging/Channel.h

```diff
@@ -0,0 +1,195 @@
+ *      connectionless transport, channel will contain peer address and session
+ *      key; when using connection oriented transport, channel will contain
+ *      connection handle and session key.
+ *
+ *      Channel is not session. Session do persistent through cold reboot, but
+ *      channel doesn't. Applications must re-establish channels after a cold
+ *      reboot.
+ *
+ *      Because channel is a local concept, peer device is not able aware of
+ *      channel establishment events. Instead, peer device is able to aware
+ *      session establishment events, connection establishment events for
+ *      connection oriented transport and message received events for
+ *      connectionless transport.
+ */
+
+#pragma once
+
+#include <transport/raw/MessageHeader.h>
+
+namespace chip {
+
+/**
+ *  @brief
+ *    The ChannelBuilder object provides information to build a Channel.
+ *
+ *    ChannelBuilder can be used by application to provide information to
+ *    request a channel to a peer. When a channel is requested via
+ *    ExchangeManager::EstablishChannel, a ChannelHandle will be return to
+ *    represent the state of the channel
+ *
+ *    The ChannelBuilder object should be short-live and it is only used within
+ *    ExchangeManager::EstablishChannel call, after the call its state will be
+ *    copied into an internal object represented by ChannelHandle, then the
+ *    ChannelBuilder can be safely released.
+ */
+class ChannelBuilder
+{
+public:
+    enum class NetworkPreference
+    {
+        kNetwork_Default,
+        kNetwork_HighBandwidth,
+        kNetwork_LowLatency,
+        kNetwork_LowPowerConsumption,
+    };
+
+    enum class TransportPreference
+    {
+        kTransport_Connectionless,           // default
+        kTransport_PreferConnectionOriented, // will fallback to connectionless TCP is not supported
+        kTransport_ConnectionOriented,       // will fail if TCP is not supported
+    };
+
+    enum class SessionType
+    {
+        kSession_NoEncryption,
+        kSession_PASE, // Use SPAKE2 key exchange
+        kSession_CASE, // Use SIGMA key exchange
+    };
+
+    ChannelBuilder & SetPeerNodeId(NodeId peerNodeId)
+    {
+        mPeerNodeId = peerNodeId;
+        return *this;
+    }
+    NodeId GetPeerNodeId() { return mPeerNodeId; }
+
+    ChannelBuilder & SetNetworkPreference(NetworkPreference preference)
+    {
+        mNetworkPreference = preference;
+        return *this;
+    }
+    NetworkPreference GetNetworkPreference() { return mNetworkPreference; }
+
+    ChannelBuilder & SetTransportPreference(TransportPreference preference)
+    {
+        mTransportPreference = preference;
+        return *this;
+    }
+    TransportPreference GetTransportPreference() { return mTransportPreference; }
+
+    ChannelBuilder & SetSessionType(SessionType preference)
+    {
+        mSessionType = preference;
+        return *this;
+    }
+    SessionType GetSessionType() { return mSessionType; }
```

Comment:
nit, but I prefer getters to omit the word `Get()`

Reply:
It conflicts with enum class name if without `Get`.